connectionmanager/connectionmanager.go

```diff
@@ -0,0 +1,118 @@
+package connectionmanager
```

Comment:
connectionmanager feels like an eyesore to me, suggest `package connection` if it doesn't conflict with anything else?

Reply:
I've fought a lot with this name. Happy for something better!

Connection isn't quite accurate since this manages the initialization/shutdown of any number of items as long as they support the initializer/closer interfaces. Indeed, connection is probably even wrong, though it nods to the intent of this managing things like database/elastic search connections. It provides a method of bringing them up and shutting them down as a group, rather than individually. I don't have a good name, so I stuck with connection manager. The file used to be called connectionmgr, but Levi set me straight on short names.

Reply:
`connectiongroup`? Naming is hard, I'll come back to it if I can think of something...